os/kernel/sched/sched_processtimer.c

```diff
@@ -215,6 +215,38 @@ static inline void sched_process_scheduler(void)
 #  define sched_process_scheduler()
 #endif
 
+/************************************************************************
+ * Name:	idle_panic_timer
+ *
+ * Description:
+ *  This function check if the system is idle for a long time. It
+ *  will panic the system if the system if so.
+ * 
+ * Limitation:
+ * 	This function can detect only those cases where a high-priority thread 
+ *  continuously holds the CPU. SMP case is not considered
+ *
+ * Inputs:
+ *   None
+ *
+ * Return Value:
+ *   None
+ *
+ ************************************************************************/
+void idle_panic_timer(void)
+{
+	static int last_pid = -1;
+	static int timer_count = 0;
+	int pid = getpid();
+	
+	if (last_pid != pid) {
+		last_pid = pid;
+		timer_count = 0;
+	} else if (++timer_count >= CONFIG_IDLE_PANIC_TIMER * TICK_PER_SEC) {
+		PANIC();
```

Comment:
Let us use DEBUGPANIC_INFO instead of PANIC and print the reason for calling panic. `PANIC called because device hang detection kicked in after %d secs`